pkg/transfer/flex/main.go

```diff
@@ -128,91 +154,385 @@ func (volp *DatasetVolumes) readDatasetOpts(mountPath string) (*datasetOpts, err
 	dec := json.NewDecoder(f)
 	err = dec.Decode(dsopts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode options")
+		return nil, errors.Wrap(err, "failed to decode metadata")
 	}
 
 	return dsopts, nil
 }
 
-func (volp *DatasetVolumes) deleteDatasetOpts(mountPath string) error {
-	path := filepath.Join(mountPath, "..", filepath.Base(mountPath)+".json")
-	return os.Remove(path)
+func (volp *DatasetVolumes) deleteDatasetOpts(path string) error {
+	err := os.Remove(path)
+	return errors.Wrap(err, "failed to delete metadata file")
 }
 
-//Init the flex volume
-func (volp *DatasetVolumes) Init() (Capabilities, error) {
-	return Capabilities{Attach: false}, nil
+//Creates a file with a file system inside of it that can be mounted
+func (volp *DatasetVolumes) createFSInFile(path string, filesystem FileSystem, size int64) error {
+	//Create file with room to contain writable file system
+	f, err := os.Create(path)
+	if err != nil {
+		err = errors.Wrap(err, "failed to create file system file")
```

Comment:
To have more consistency, you can use one-liners:
` return errors.Wrap(err, "failed to create file system file")`

Reply:
Fixed, was a leftover of something I tried with `defer`